tests/components/openweathermap/test_config_flow.py

```diff
@@ -0,0 +1,216 @@
+"""Define tests for the OpenWeatherMap config flow."""
+from asynctest import MagicMock, patch
+from pyowm.exceptions.api_call_error import APICallError
+from pyowm.exceptions.api_response_error import UnauthorizedError
+
+from homeassistant import data_entry_flow
+from homeassistant.components.openweathermap.const import (
+    CONF_LANGUAGE,
+    DEFAULT_FORECAST_MODE,
+    DEFAULT_LANGUAGE,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+)
+
+from tests.common import MockConfigEntry
+
+CONFIG = {
+    CONF_NAME: "openweathermap",
+    CONF_API_KEY: "foo",
+    CONF_LATITUDE: 50,
+    CONF_LONGITUDE: 40,
+    CONF_MODE: DEFAULT_FORECAST_MODE,
+    CONF_LANGUAGE: DEFAULT_LANGUAGE,
+}
+
+VALID_YAML_CONFIG = {CONF_API_KEY: "foo"}
+
+
+async def test_form(hass):
+    """Test that the form is served with valid input."""
+    mocked_owm = _create_mocked_owm(True)
+
+    with patch(
+        "homeassistant.components.openweathermap.config_flow.OWM",
+        return_value=mocked_owm,
+    ), patch(
+        "homeassistant.helpers.update_coordinator.DataUpdateCoordinator.async_refresh",
+        _async_refresh,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == SOURCE_USER
+        assert result["errors"] == {}
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        )
+
+        await hass.async_block_till_done()
+
+        conf_entries = hass.config_entries.async_entries(DOMAIN)
+        entry = conf_entries[0]
+        assert entry.state == "loaded"
+
+        await hass.config_entries.async_unload(conf_entries[0].entry_id)
+        await hass.async_block_till_done()
+        assert entry.state == "not_loaded"
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == CONFIG[CONF_NAME]
+        assert result["data"][CONF_LATITUDE] == CONFIG[CONF_LATITUDE]
+        assert result["data"][CONF_LONGITUDE] == CONFIG[CONF_LONGITUDE]
+        assert result["data"][CONF_API_KEY] == CONFIG[CONF_API_KEY]
+
+
+async def test_form_import(hass):
+    """Test we can import yaml config."""
+    mocked_owm = _create_mocked_owm(True)
+
+    with patch(
+        "homeassistant.components.openweathermap.config_flow.OWM",
+        return_value=mocked_owm,
+    ), patch(
+        "homeassistant.helpers.update_coordinator.DataUpdateCoordinator.async_refresh",
+        _async_refresh,
```

Comment:
This shouldn't be needed.  If it is, please patch the library instead of DataUpdateCoordinator